packages/visx-responsive/src/types/index.ts

```diff
@@ -23,3 +23,5 @@ interface ResizeObserverPolyfill {
 }
 
 export { ResizeObserver, ResizeObserverCallback, ResizeObserverPolyfill };
+
+export type Simplify<T> = { [Key in keyof T]: T[Key] } & {};
```

Comment:
curious why this was needed (open to it if it gets things to work nicely, just harder to tell why from the github UI)

Reply:
I like to put it in library code types for better DX for end users. So the HOC has a return type of `React.ComponentType<WithParentSizeComponentProps<P>>` and without `Simplify` in `WithParentSizeComponentProps<P>` when you hover over a component wrapped in the HOC in the IDE you'd see something like
 ```
const Component: React.ComponentType<WithParentSizeComponentProps<ComponentProps>>
```

With `Simplify`, you'll see 
```
const Component: React.ComponentType<{
    role: string;
    debounceTime?: number | undefined;
    enableDebounceLeadingCall?: boolean | undefined;
    initialWidth?: number | undefined;
    initialHeight?: number | undefined;
}>
```

(`role` is the component's own prop)